vivisect/tests/testswitches.py

```diff
@@ -0,0 +1,187 @@
+    ]
+
+
+def applyMaps(vw, maps=MS_MAPS, mapbase=0x400000, bufferpages=2):
+    bufferpgsz = bufferpages * 4096
+    vw.addMemoryMap(mapbase - bufferpgsz, e_mem.MM_RWX, 'testswitches', '@' * bufferpgsz)
+
+    mapdelta = mapbase - maps[0][0]
+    
+
+    vw.addFile('testswitches', mapbase, "@@@@@@@@")
+    for mapva, sz, perms, mem in maps:
+        mapva += mapdelta
+        vw.addMemoryMap(mapva, perms, 'testswitches', mem)
+        vw.addSegment(mapva, len(mem), 'switch_code_%x' % mapva, 'testswitches')
+
+
+def genMsSwitchWorkspace(maps=MS_MAPS, mapbase=0x400000, bufferpages=2):
+    vw = vivisect.VivWorkspace()
+    vw.setMeta('Architecture', 'i386')
+    vw.setMeta('Platform', 'windows')
+    vw.setMeta('Format', 'pe')
+    vw._snapInAnalysisModules()
+
+    applyMaps(vw, maps, mapbase, bufferpages)
+    return vw
+
+def genGccSwitchWorkspace(maps=LIBC_MAPS, mapbase=0x400000, bufferpages=2):
+    vw = vivisect.VivWorkspace()
+    vw.setMeta('Architecture', 'i386')
+    vw.setMeta('Platform', 'linux')
+    vw.setMeta('Format', 'elf')
+    vw._snapInAnalysisModules()
+
+    applyMaps(vw, maps, mapbase, bufferpages)
+    return vw
+
+cbs_ms_0 = [  
+        (4194304, 13, 4194304),
+        (4194317, 9, 4194304),
+        (4194326, 9, 4194304),
+        (4194335, 38, 4194304),
+        (4194373, 14, 4194304),
+        (4194387, 10, 4194304),
+        (4244306, 12, 4194304),
+        (4244318, 27, 4194304),
+        (4274127, 6, 4194304),
+        (4274133, 9, 4194304),
+        (4274142, 9, 4194304),
+        (4274151, 5, 4194304),
+        (4274156, 14, 4194304),
+        (4308616, 17, 4194304),
+        (4308633, 26, 4194304),
+        (4308659, 26, 4194304),
+        (4308685, 26, 4194304),
+        (4308711, 15, 4194304)]
+
+class MsSwitchTest(unittest.TestCase):
+    def test_ms_switch_0(self):
+        vw = genMsSwitchWorkspace(MS_MAPS, 0x400000)
+        vw.makeFunction(0x400000)
+        
+        self.assertEqual(vw.getFunctionBlocks(0x400000), cbs_ms_0)
+        #self.assertEqual(vw.getXrefsFrom(0x
+
+
+
+
+
+class LibCSwitchTest(unittest.TestCase):
+    def test_libc_switch_0(self):
+        vw = genGccSwitchWorkspace(LIBC_MAPS, 0x500000)
```

Comment:
This doesn't actually assert anything.

Reply:
yes.  still working on getting the tests completed.